### PR TITLE
refactor(core): Switch plain errors in `cli` to `ApplicationError`

### PR DESCRIPTION
**Pull request by ivov**
Ensure all errors in `cli` are `ApplicationError` or children of it and contain no variables in the message, to continue normalizing all the errors we report to Sentry

Follow-up to: https://github.com/n8n-io/n8n/pull/7839

### CODE REVIEW

File: packages/cli/src/environments/sourceControl/sourceControlPreferences.service.ee.ts [hunk 1]
```diff
@@ -150,7 +150,9 @@ export class SourceControlPreferencesService {
 			validationError: { target: false },
 		});
 		if (validationResult.length > 0) {
-			throw new Error(`Invalid source control preferences: ${JSON.stringify(validationResult)}`);
+			throw new ApplicationError('Invalid source control preferences', {
+				extra: { preferences: JSON.stringify(validationResult) },
```
Comment (netroy) [inline packages/cli/src/environments/sourceControl/sourceControlPreferences.service.ee.ts:154]: don't need to do a `JSON.stringify` here anymore.

File: packages/cli/src/services/dynamicNodeParameters.service.ts [hunk 2]
```diff
@@ -182,8 +183,9 @@ export class DynamicNodeParametersService {
 	) {
 		const method = nodeType.methods?.[type]?.[methodName];
 		if (typeof method !== 'function') {
-			throw new Error(
-				`The node-type "${nodeType.description.name}" does not have the method "${methodName}" defined!`,
+			throw new ApplicationError(
+				`Node type "${nodeType.description.name}" does not have the method "${methodName}" defined!`,
```
Comment (netroy) [inline packages/cli/src/services/dynamicNodeParameters.service.ts:187]: maybe we should have nodeType and methodName only in tags, and keep the message string constant.

**Review by netroy — CHANGES_REQUESTED** [review]
Thanks for doing this 🙏🏽

File: packages/cli/src/services/dynamicNodeParameters.service.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/services/dynamicNodeParameters.service.ts]: > keep the message string constant

Thank you - definitely been aiming for this but missed this spot. Kept `nodeType` and `credentialType` for `tags` and all the rest for `extra`.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.19.4`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.19.4)

**Pull request merged**